sdk/program/src/entrypoint.rs

```diff
@@ -123,6 +123,36 @@ pub const NON_DUP_MARKER: u8 = u8::MAX;
 /// ```
 #[macro_export]
 macro_rules! entrypoint {
+    ($process_instruction:ident) => {
+        /// # Safety
+        #[no_mangle]
+        pub unsafe extern "C" fn entrypoint(input: *mut u8) -> u64 {
+            let (program_id, accounts, instruction_data) =
+                unsafe { $crate::entrypoint::deserialize(input) };
+            match $process_instruction(&program_id, &accounts, &instruction_data) {
+                Ok(()) => $crate::entrypoint::SUCCESS,
+                Err(error) => error.into(),
+            }
+        }
+        $crate::custom_heap_default!();
+        $crate::custom_panic_default!();
+    };
+}
+
+/// Declare the program entrypoint and set up global handlers.
+///
+/// This is similar to the `entrypoint!` macro, except that it does not perform
+/// any dynamic allocations, and instead writes the input accounts into a pre-
+/// allocated array.
+///
+/// This version reduces compute unit usage by 20-30 compute units per unique
+/// account in the instruction. It may become the default option in a future
+/// release.
+///
+/// For more information about how the program entrypoint behaves and what it
+/// does, please see the documentation for [`entrypoint!`].
+#[macro_export]
```

Comment:
Can we just mention the 64 account limit? It's implied for now, but whenever the feature is one day activated, at least the documentation here will let people know of the limit.

Reply:
Sure! Done in [00d87de](https://github.com/anza-xyz/agave/pull/2618/commits/00d87de6e3f75c4fb2a049b6603025620624fa5e)